Xamarin.Forms.Core/Application.cs

```diff
@@ -106,8 +110,22 @@ internal override ReadOnlyCollection<Element> LogicalChildrenInternal
 			get { return _logicalChildren ?? (_logicalChildren = new ReadOnlyCollection<Element>(InternalChildren)); }
 		}
 
+		[ThreadStatic]
+		static NavigationProxy _navigationProxy;
```

Comment:
@wachs Do you remember why these changes were made to NavigationProxy?

Reply:
This changes was made for navigation happen separately between windows. If you do not have this ThreadStatic, the navigation of the first window is replaced by the second.

Reply:
But native Forms embedding only supports ContentPage, so there shouldn't be any navigation in the second window.

Reply:
You're right! In a previous deployment we were able to implement any type, for example a MasterDetailPage. Once we switch to Dependency Injection it is no longer possible.

Reply:
I will make the adjustment!